lib/src/convert_es6_proxy.html

```diff
@@ -0,0 +1,198 @@
+<!--
+Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+for details. All rights reserved. Use of this source code is governed by a
+BSD-style license that can be found in the LICENSE file.
+-->
+<link rel="import" href="convert.html">
+<script>
+  Polymer.PolymerInterop = Polymer.PolymerInterop || {};
+  Polymer.PolymerInterop.ES6 = Polymer.PolymerInterop.ES6 || {};
+
+  (function(ES6) {
+
+    ES6.Unsupported = {};
+
+    var _throwShouldBeOverridden = function() {
+      throw new Error("This method should be overridden by dart implementation");
+    };
+
+    ES6._dartArrayGet = _throwShouldBeOverridden;
+    ES6._dartArrayPut = _throwShouldBeOverridden;
+    ES6._dartArrayLength = _throwShouldBeOverridden;
+    ES6._dartArraySetLength = _throwShouldBeOverridden;
+
+    ES6._dartGet = _throwShouldBeOverridden;
+    ES6._dartPut = _throwShouldBeOverridden;
+    ES6._dartKeys = _throwShouldBeOverridden;
+    ES6._dartMapDelete = _throwShouldBeOverridden;
+
+    /**
+     * List ES6JsProxy support
+     */
+
+    var arrayCommonBehaviors = {
+      '__dartClass__' : function(instance) { return instance; },
+      '__isES6Proxy__' : function(instance) { return true; },
+      'toString' : function(instance) { return function() { return "Array["+ES6._dartArrayLength(instance)+"]"; }; },
+      'length' : function(instance) { return ES6._dartArrayLength(instance); }
+    };
+
+    // TODO(dam0vment) investigate if overriding only get/setOwnPropertyDescriptor and ownKeys
+    // is enough (all the other methods could just depend on those)
+    ES6.createES6JsProxyForArray = function(instance) {
+      return new Proxy([],{
+        'get' : function(target,propertyName,proxy) {
+
+          if (arrayCommonBehaviors.hasOwnProperty(propertyName)) {
+            return arrayCommonBehaviors[propertyName].call(this,instance);
+          }
+
+          if (isNaN(propertyName.toString())) {
+            return target[propertyName];
+          }
+
+          return ES6._dartArrayGet(instance,+propertyName);
+        },
+
+        'has' : function(target,prop) {
+          var len = ES6._dartArrayLength(instance);
+          var pos = +prop;
```

Comment:
eh, you know js people are very bad guys ? they tend to put all sort of things inside an array. Dart people too will add some symbol to store some caching info inside it. So I can't assume that property is a number.